### PR TITLE
feat(installer): wire the corporate CA into cosign/helm/git, not just curl (#583)

### PR DESCRIPTION
**Pull request by shujaatTracebloc**
## What & why

Closes #583 (child 2/4 of #578, network resilience — **the single biggest lever** for enterprise networks).

A TLS-inspecting corporate proxy re-signs HTTPS with a corporate root CA. Tools that don't trust that root fail x509 — the class behind **both** field failures (the k3d-checksum failure and the cosign/sigstore verification failure). curl already honored `CURL_CA_BUNDLE`, and the **k3d nodes** got the CA at cluster-create (#424), but **cosign, helm, and git got nothing**. This PR extends the *same* resolved CA to every host tool that doesn't inherit the system trust store on its own.

## Changes

**Bootstrap — cosign** (`install.sh`, `install.ps1`; not manifested — the trust root)
- Before cosign runs, export `SSL_CERT_FILE` from `TRACEBLOC_CA_BUNDLE` / `CURL_CA_BUNDLE`. cosign's Go HTTPS client reads `SSL_CERT_FILE` (added to the roots on modern Go, incl. Windows), so keyless verification's calls trust the corporate CA instead of failing x509. No-op when unset.

**Main installer — helm / git / curl** (`cluster.sh` `wire_ca_trust`, `install-k8s.ps1` `Set-ToolTrust`)
- Export `SSL_CERT_FILE` + `GIT_SSL_CAINFO` (+ `CURL_CA_BUNDLE`) from the resolved bundle, **before** preflight's HTTPS probes and any tool download, so helm, git and curl all trust it end-to-end.
- One plain-language line: `Trusting your company's certificate for cosign, helm, git and downloads.`
- Idempotent; no-op when unconfigured; **fails fast** (before any privileged step) on a set-but-unreadable bundle — reusing the existing `_resolve_ca_bundle` / `Resolve-CaBundle` contract.

## On the acceptance criteria
- **Inspected-but-not-blocked networks complete, verified:** when a CA is provided it's trusted by every tool; when it isn't, tools fall back to the **system store** (which enterprise IT usually populates) — so the common case needs **no user knowledge of CAs**.
- **Cross-platform, one-to-one:** `wire_ca_trust` ↔ `Set-ToolTrust`, bootstrap wiring on both.
- **PII-free output:** the announce names no paths or secrets.
- The **k3d nodes** remain trusted separately at cluster-create (#424, unchanged).

## Tests
- **bash:** `wire_ca_trust` exports the three vars / no-ops when unset / hard-fails on an unreadable bundle (cluster.bats); the **bootstrap** integration test runs the real bootstrap with `TRACEBLOC_CA_BUNDLE` set and a recording mock cosign, asserting cosign saw `SSL_CERT_FILE` (install-bootstrap.bats).
- **PowerShell:** `Set-ToolTrust` exports `SSL_CERT_FILE`/`GIT_SSL_CAINFO` / no-ops when unset (install-k8s.Tests.ps1); the bootstrap sets `SSL_CERT_FILE` from the CA env (install.Tests.ps1).
- Full Pester 436/0/9; cluster.bats 78/0; manifest `--check`, check-facts, check-style clean. (`cluster.sh` / `install-k8s.*` are manifested — `manifest.sha256` regenerated; the bootstrap scripts are the trust root and stay unmanifested.)

## Notes
- Builds on #582's detection (child 1/4) but ships independently — this is the *acting* layer.
- The offline-Rekor / expired-cert cosign case (no live sigstore) is the separate **#584** offline-bundle track.
- The one local test failure anywhere is the pre-existing macOS-only `install-bootstrap` "early bailout" flake (passes on CI Linux; untouched here).

---

> [!NOTE]
> **Medium Risk**
> Changes installer TLS trust wiring across bootstrap and main install paths on all platforms; behavior is platform-gated and fail-fast, but mistakes could still affect cosign/git/helm behind corporate proxies.
> 
> **Overview**
> **Corporate CA trust is applied earlier and more honestly across the installer stack (#583), so TLS-inspecting proxies fail at setup with clear errors instead of mid-download x509 failures.**
> 
> On **Linux**, `wire_ca_trust` (bash) sets `SSL_CERT_FILE` and `GIT_SSL_CAINFO` from `TRACEBLOC_CA_BUNDLE` / `CURL_CA_BUNDLE` only when those vars are unset; it does **not** overwrite `CURL_CA_BUNDLE`. The main `install-k8s` flow calls this **before** preflight and tool downloads. Bootstrap `install.sh` adds `_bootstrap_wire_ca` so **cosign** sees `SSL_CERT_FILE` on Linux before manifest verification.
> 
> On **macOS**, the same helpers **validate** the bundle path but **export nothing**—Go tools use Keychain, and exporting could shrink curl trust—users get Keychain/offline hints instead.
> 
> On **Windows**, `Set-ToolTrust` sets **only** `GIT_SSL_CAINFO` (not `SSL_CERT_FILE`); `install.ps1` validates CA readability before cosign but likewise does not wire PEM into Go tools, with hints to use the certificate store or offline installer.
> 
> **User messaging** distinguishes what was wired vs pre-set bundles kept (`replace-not-augment`), avoiding false “Trusting…” success when nothing changed. Tests cover Linux/macOS bootstrap behavior, `wire_ca_trust` edge cases, and PowerShell `Set-ToolTrust`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1b40cbe773277d3504a483494ea8eb981f5f5908. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -330,9 +330,24 @@ _sha256_of() {
...
+# Wire an explicitly-provided corporate CA into cosign (SSL_CERT_FILE) so a
+# TLS-inspecting proxy that re-signs HTTPS doesn't fail the signature check with an
+# x509 error (#583) — the class behind the field TLS-inspection failures. curl
+# already honors CURL_CA_BUNDLE; this adds cosign, whose Go HTTPS client reads
+# SSL_CERT_FILE. No-op when unset; the full host-tool wiring (helm/git) is done by
+# the main installer's wire_ca_trust once the libs are fetched + verified.
+_bootstrap_wire_ca() {
+  local ca="${TRACEBLOC_CA_BUNDLE:-${CURL_CA_BUNDLE:-}}"
+  [[ -n "$ca" && -f "$ca" && -r "$ca" ]] || return 0
+  export SSL_CERT_FILE="$ca" CURL_CA_BUNDLE="$ca"
```
[the remark this answers is not in the document]
Comment (shujaatTracebloc) [reply scripts/install.sh]: Fixed in b4c3a6d. You're right — `CURL_CA_BUNDLE` is replace-not-augment, and `TRACEBLOC_CA_BUNDLE` is typically a corp-root-only PEM (its k3d-node use), so re-exporting it could strip the public roots curl needs. All three sites (`_bootstrap_wire_ca`, `wire_ca_trust`, `Set-ToolTrust`) now set only `SSL_CERT_FILE` (+ `GIT_SSL_CAINFO` in the main installer) — curl keeps honoring the user's own `CURL_CA_BUNDLE` natively and we never overwrite it. That also makes the bash and PS bootstraps symmetric (both set only `SSL_CERT_FILE`). Added a regression test asserting `wire_ca_trust` leaves a pre-set `CURL_CA_BUNDLE` intact while still wiring `SSL_CERT_FILE` for cosign/helm/git.

File: scripts/install.ps1 [hunk 2]
```diff
@@ -319,6 +319,14 @@ function Confirm-ManifestSignature {
     [string]$TmpDir,
     [bool]$AllowUnverified
   )
+  # Wire an explicitly-provided corporate CA into cosign (SSL_CERT_FILE) before its
+  # HTTPS calls, so a TLS-inspecting proxy that re-signs HTTPS doesn't fail the
+  # signature check with an x509 error (#583). Invoke-WebRequest uses the system
+  # store; cosign's Go client reads SSL_CERT_FILE (added to the roots on modern Go,
+  # incl. Windows). No-op when unset; helm/git wiring happens in the main installer.
+  $ca = if ($env:TRACEBLOC_CA_BUNDLE) { $env:TRACEBLOC_CA_BUNDLE } elseif ($env:CURL_CA_BUNDLE) { $env:CURL_CA_BUNDLE } else { $null }
+  if ($ca -and (Test-Path -LiteralPath $ca -PathType Leaf)) { $env:SSL_CERT_FILE = $ca }
```
[the remark this answers is not in the document]
Comment (shujaatTracebloc) [reply scripts/install.ps1]: You're right — Go reads SSL_CERT_FILE only on Linux; on Windows it uses the cert store and on macOS the Keychain, so my SSL_CERT_FILE wiring was inert for cosign/helm there and my comment was wrong. Fixed in bdb2bf9: Linux keeps SSL_CERT_FILE (effective); macOS still sets it but the announce names only git + downloads and hints that cosign/helm read the Keychain; Windows sets only GIT_SSL_CAINFO (Git-for-Windows is OpenSSL-backed) and hints cosign/helm read the cert store; the Windows bootstrap drops the inert SSL_CERT_FILE entirely. The robust cross-platform cosign fix for a PEM-only CA is the offline bundle (#584). Tests updated for the platform-accurate behavior.

File: scripts/install.sh [hunk 3]
```diff
@@ -330,9 +330,26 @@ _sha256_of() {
...
+# x509 error (#583) — the class behind the field TLS-inspection failures. Only
+# SSL_CERT_FILE (cosign's Go HTTPS client reads it): curl ALREADY honors the user's
+# own CURL_CA_BUNDLE natively, and we must NOT re-export it from a corp-root-only
+# TRACEBLOC_CA_BUNDLE — CURL_CA_BUNDLE is replace-not-augment, so that would drop the
+# public roots curl needs for the manifest/sig fetches (Bugbot). No-op when unset;
+# helm/git wiring is done by the main installer's wire_ca_trust.
+_bootstrap_wire_ca() {
+  local ca="${TRACEBLOC_CA_BUNDLE:-${CURL_CA_BUNDLE:-}}"
+  [[ -n "$ca" && -f "$ca" && -r "$ca" ]] || return 0
+  export SSL_CERT_FILE="$ca"
```
[the remark this answers is not in the document]
Comment (shujaatTracebloc) [reply scripts/install.sh]: Fixed in bdb2bf9. Both bootstraps now fail fast with a clear "…can't be read" message when TRACEBLOC_CA_BUNDLE/CURL_CA_BUNDLE is set but missing/unreadable, before cosign runs — so the operator gets the actionable error instead of a later generic authenticity failure. Added regression tests (bash run_boot + a PS source assertion).

File: scripts/lib/cluster.sh [hunk 4]
```diff
@@ -142,6 +142,32 @@ _resolve_ca_bundle() {
...
+# fails fast on a set-but-unreadable bundle. The announce names only what actually
+# takes effect on this platform.
+wire_ca_trust() {
+  local ca rc=0
+  ca="$(_resolve_ca_bundle)" || rc=$?
+  if [[ "$rc" -eq 2 ]]; then
+    error "$ca is set but its CA bundle file can't be read — fix its path/permissions and re-run."
+  fi
+  [[ -z "$ca" ]] && return 0
+  export SSL_CERT_FILE="$ca" GIT_SSL_CAINFO="$ca"
```
[the remark this answers is not in the document]
Comment (shujaatTracebloc) [reply scripts/lib/cluster.sh]: Fixed in f45ae33. Same replace-not-augment contract as CURL_CA_BUNDLE — you're right it applies to GIT_SSL_CAINFO (and SSL_CERT_FILE) too. Now every trust var is set only when the user hasn't already set it: a pre-set GIT_SSL_CAINFO / SSL_CERT_FILE is left intact (their fuller bundle wins), while an unset one is still wired from the corp CA. Added bash + Pester regression tests asserting a pre-set bundle survives.

File: scripts/install-k8s.ps1 [hunk 5]
```diff
@@ -1810,6 +1810,23 @@ function Resolve-CaBundle {
...
+# (replace-not-augment). The k3d nodes are trusted at cluster-create (#424). No-op when
+# unconfigured; Resolve-CaBundle fails fast on an unreadable bundle.
+function Set-ToolTrust {
+  $ca = Resolve-CaBundle
+  if (-not $ca) { return }
+  # Don't clobber a fuller pre-set GIT_SSL_CAINFO (replace-not-augment): only set it
+  # when the user hasn't already (Bugbot).
+  if (-not $env:GIT_SSL_CAINFO) { $env:GIT_SSL_CAINFO = $ca }
+  Ok "Trusting your company's certificate for git and downloads."
+  Hint "On Windows, cosign and helm read the certificate store, not a PEM file - import your corporate CA into Cert:\LocalMachine\Root (or use the offline installer) so they trust it too."
```
[the remark this answers is not in the document]
Comment (shujaatTracebloc) [reply scripts/install-k8s.ps1]: Fixed in d07588f. The announce now names only what's actually wired: Windows says "…for git." (only GIT_SSL_CAINFO), and the store hint covers cosign, helm AND the installer's downloads (Invoke-WebRequest/Schannel read the cert store). Dropped the same "downloads" over-claim on Linux/macOS too — curl downloads trust the user's own CURL_CA_BUNDLE, which we don't touch, so Linux announces "cosign, helm and git" and macOS "git" only.

File: scripts/install.ps1 [hunk 6]
```diff
@@ -319,6 +319,17 @@ function Confirm-ManifestSignature {
...
+  # An explicit corporate CA can't be wired into cosign via env on Windows: cosign is
+  # Go, and Go on Windows reads the certificate store and IGNORES SSL_CERT_FILE
+  # (Bugbot). So there's nothing to export here — the CA must live in the Windows
+  # store (Cert:\LocalMachine\Root), or use the offline installer path (#584).
+  # We still validate the path so a typo fails fast with a clear message rather than
+  # a later generic cosign authenticity error.
+  $ca = if ($env:TRACEBLOC_CA_BUNDLE) { $env:TRACEBLOC_CA_BUNDLE } elseif ($env:CURL_CA_BUNDLE) { $env:CURL_CA_BUNDLE } else { $null }
+  if ($ca -and -not (Test-Path -LiteralPath $ca -PathType Leaf)) {
+    throw "A CA bundle is set (TRACEBLOC_CA_BUNDLE/CURL_CA_BUNDLE) but '$ca' can't be read - fix its path and re-run."
+  }
```
[the remark this answers is not in the document]
Comment (shujaatTracebloc) [reply scripts/install.ps1:337]: Fixed in d07588f. The Windows bootstrap now also opens the file ([System.IO.File]::OpenRead) after the existence check, mirroring bash's -r and Resolve-CaBundle — a present-but-unreadable CA fails fast here with a clear "can't be read" message instead of a later generic cosign authenticity error.

**Review by LukasWodka — APPROVED** [review]
